plasmapy/dispersion/tests/test_api.py

```diff
@@ -0,0 +1,85 @@
+import astropy.units as u
+from astropy import constants
+import pytest
+
+@pytest.mark.xfail(strict=True)
+def test_dispersion_api():
+    """
+    This is a broad overview of the initial ideas we had on the sample API for the dispersion solver
+    as per https://github.com/PlasmaPy/plasmapy-meeting-notes/blob/master/2019-09-24-design-meeting.md
+
+    It is marked as xfail - ideally a later PR will cause this 
+    """
+
+    p = Particle('p')
+
+    # class Species(Particle):
+    #     def __init__(self, particle_name, temperature, anisotropy, density, drift):
+    #         ...
+        
+    from plasmapy.atomic import CustomParticle # probably the best place for it but negotiable
+    custom_ion = CustomParticle(mass=25*electron_mass, charge=2 * constants.e.si)
+
+    from plasmapy.dispersion import Species # place negotiable? better name?
+
+    species0 = Species(custom_ion, 
+        anisotropy = 1/3.1415,
+        # or solve for anisotropy from T_perpendicular, T_parallel, but for now - it's fine to supply it
+        density = 1e23 * u.m**-3,
+        # in general, example: drift = [0, 0, 2] * (u.m/u.s), but for now don't worry about direction
+        drift = 2 * (u.m/u.s),
+    )
+
+    # physics initialization 
+    from plasmapy.dispersion import PhysicalInputs #, DispersionSolver
+    dispersion_input = PhysicalInputs(species = [species0],
+        magnetic_field = u.Quantity([0, 0, 5], u.T),
+        # kwargs:
+        wavenumber = np.linspace(0, 10, 1000) / u.m,
+        # wavenumber = np.array([0.1, 10, 3,]) / u.m,
+        angle = np.linspace(0, np.pi, 1000),
+        # v_Alfven/c needs to be controlled - magnetic_field, density give us v
+        # TODO what do we need for the relativistic dispersion?
+        # TODO Ask Daniel Verscharen and other astrophysicists
+        )    
+
+    # class DispersionSolver:
+    #     solvers = {'kinetic': KineticDispersionSolver,
+    #                'two_fluid': tfst}
+        
+    #     def __init__(self, solver_type, **solver_kwargs):
+    #         # doc issues but easy to extend
+        
+    from plasmapy.dispersion import KineticDispersionSolver
+    kds = KineticDispersionSolver(dispersion_input,
+                                  number_of_iterations = 10000,         
```

Comment:
These numerical parameters were directly copied from Daniel Verscharen's NHDS code. Depending on whether we write our own or if we use a compiled version of Daniel's code, we might need to change these.